### PR TITLE
avalanche support

### PR DESCRIPTION
**Pull request by dereknelson**
Fixes APP-1089

## What changed (plus any additional context for devs)
got a new network in the network tab!

## Screen recordings / screenshots
image

## What to test
~at the moment, test why there are no routes for mainnet eth to avax usdc~

~in a commit or two, test that the swaps actually go thru~

swap some avax & browse some avax nfts?

### CODE REVIEW

File: src/components/gas/GasSpeedButton.js [hunk 1]
```diff
@@ -348,6 +348,7 @@ const GasSpeedButton = ({
       case networkTypes.arbitrum:
       case networkTypes.zora:
       case networkTypes.base:
+      case networkTypes.avalanche:
```
Comment (skylarbarrera) [inline src/components/gas/GasSpeedButton.js:351]: avalanche native is AVAX

Comment (dereknelson) [reply src/components/gas/GasSpeedButton.js:351]: good catch ty

**Comment by dereknelson** [comment]
swaps failing with a server error 
image

``` 
LOG  [10:23:57 PM] ⚡⚡⚡  Rap blew up [Error: missing response (requestBody="{\"method\":\"eth_getTransactionCount\",\"params\":[\"0x2e67869829c734ac13723a138a952f7a8b56e774\",\"pending\"],\"id\":71,\"jsonrpc\":\"2.0\"}", requestMethod="POST", serverError={}, url="undefined/1/undefined", code=SERVER_ERROR, version=web/5.7.1)]
 LOG  [10:23:57 PM] [3 INNER] error running action, code: SERVER_ERROR
 LOG  [10:23:57 PM] [4 INNER] displaying error message Oh no! There was a problem submitting the transaction. Please try again.
```

seems like the quote is good (filtering out some contents for brevity): 

```
FILLCROSSCHAINSWAP {"allowanceTarget": "0x3a23F943181408EAC424116Af7b7790c94Cb97a5", "buyAmount": "64138591952900000", "buyAmountDisplay": "64138591952900000", "buyAmountInEth": "64138591952900000", "buyAmountMinusFees": "64138591952900000", "buyTokenAddress": "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee", "fee": "2500000000000", "feeInEth": "2500000000000", "feePercentageBasisPoints": "2500000000000000", "from": "0x2e67869829c734ac13723A138a952F7A8B56e774", "fromChainId": 1, "gasPrice": "", "inputTokenDecimals": 18, "outputTokenDecimals": 18, "protocols": [{"name": "refuel-bridge", "part": 100}], "refuel": null, "routes": [{"chainGasBalances": [Object], "fromAmount": "1000000000000000", "integratorFee": [Object], "isOnlySwapRoute": false, "maxServiceTime": 7200, "minimumGasBalances": [Object], "receivedValueInUsd": -5.397654498119517, "recipient": "0x2e67869829c734ac13723A138a952F7A8B56e774", "routeId": "de58a38f-426d-45e8-9b5f-1a49b9de1d2f", "sender": "0x2e67869829c734ac13723A138a952F7A8B56e774", "serviceTime": 60, "toAmount": "64138591952900000", "totalGasFeesInUsd": 8.027974498119518, "totalUserTx": 1, "usedBridgeNames": [Array], "userTxs": [Array]}], "sellAmount": "1000000000000000", "sellAmountDisplay": "1000000000000000", "sellAmountInEth": "1000000000000000", "sellAmountMinusFees": "997500000000000", "sellTokenAddress": "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee", "source": "socket", "swapType": "cross-chain", "to": "0x3a23F943181408EAC424116Af7b7790c94Cb97a5", "toChainId": 43114, "tradeAmountUSD": 2.92211, "tradeType": "exact_input", "txTarget": "0x3a23F943181408EAC424116Af7b7790c94Cb97a5", "value": "0x038d7ea4c68000"}
```

only thing of note is we have an undefined nonce, unless that's expected behavior
```
{"gasLimit": "0x0493e0", "maxFeePerGas": "40607905535", "maxPriorityFeePerGas": "1500000000", "nonce": undefined}
```

**Comment by dereknelson** [comment]
swaps failing with a server error 
image

``` 
LOG  [10:23:57 PM] ⚡⚡⚡  Rap blew up [Error: missing response (requestBody="{\"method\":\"eth_getTransactionCount\",\"params\":[\"0x2e67869829c734ac13723a138a952f7a8b56e774\",\"pending\"],\"id\":71,\"jsonrpc\":\"2.0\"}", requestMethod="POST", serverError={}, url="undefined/1/undefined", code=SERVER_ERROR, version=web/5.7.1)]
 LOG  [10:23:57 PM] [3 INNER] error running action, code: SERVER_ERROR
 LOG  [10:23:57 PM] [4 INNER] displaying error message Oh no! There was a problem submitting the transaction. Please try again.
```

seems like the quote is good (filtering out some contents for brevity): 

```
FILLCROSSCHAINSWAP {"allowanceTarget": "0x3a23F943181408EAC424116Af7b7790c94Cb97a5", "buyAmount": "64138591952900000", "buyAmountDisplay": "64138591952900000", "buyAmountInEth": "64138591952900000", "buyAmountMinusFees": "64138591952900000", "buyTokenAddress": "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee", "fee": "2500000000000", "feeInEth": "2500000000000", "feePercentageBasisPoints": "2500000000000000", "from": "0x2e67869829c734ac13723A138a952F7A8B56e774", "fromChainId": 1, "gasPrice": "", "inputTokenDecimals": 18, "outputTokenDecimals": 18, "protocols": [{"name": "refuel-bridge", "part": 100}], "refuel": null, "routes": [{"chainGasBalances": [Object], "fromAmount": "1000000000000000", "integratorFee": [Object], "isOnlySwapRoute": false, "maxServiceTime": 7200, "minimumGasBalances": [Object], "receivedValueInUsd": -5.397654498119517, "recipient": "0x2e67869829c734ac13723A138a952F7A8B56e774", "routeId": "de58a38f-426d-45e8-9b5f-1a49b9de1d2f", "sender": "0x2e67869829c734ac13723A138a952F7A8B56e774", "serviceTime": 60, "toAmount": "64138591952900000", "totalGasFeesInUsd": 8.027974498119518, "totalUserTx": 1, "usedBridgeNames": [Array], "userTxs": [Array]}], "sellAmount": "1000000000000000", "sellAmountDisplay": "1000000000000000", "sellAmountInEth": "1000000000000000", "sellAmountMinusFees": "997500000000000", "sellTokenAddress": "0xeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee", "source": "socket", "swapType": "cross-chain", "to": "0x3a23F943181408EAC424116Af7b7790c94Cb97a5", "toChainId": 43114, "tradeAmountUSD": 2.92211, "tradeType": "exact_input", "txTarget": "0x3a23F943181408EAC424116Af7b7790c94Cb97a5", "value": "0x038d7ea4c68000"}
```

only thing of note is we have an undefined nonce, unless that's expected behavior
```
{"gasLimit": "0x0493e0", "maxFeePerGas": "40607905535", "maxPriorityFeePerGas": "1500000000", "nonce": undefined}
```

File: src/networks/avalanche.ts [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+      speeds: [gasUtils.NORMAL, gasUtils.FAST, gasUtils.URGENT, gasUtils.CUSTOM],
+      // ?
+      gasType: 'eip1559',
+      roundGasDisplay: true,
+      OptimismTxFee: true,
+
+      // this prob can just be blockTime,
+      pollingIntervalInMs: 5_000,
+
+      getGasPrices: getBaseGasPrices,
```
Comment (skylarbarrera) [inline src/networks/avalanche.ts:53]: lot of base specific things here that still need to be updated

File: src/redux/gas.ts [hunk 3]
```diff
@@ -359,6 +362,25 @@ export const getBaseGasPrices = async () => {
   return priceData;
 };
 
+export const getAvalancheGasPrices = async () => {
+  const provider = await getProviderForNetwork(Network.avalanche);
+  const baseGasPrice = await provider.getGasPrice();
+
+  const BasePriceBumpFactor = 1.05;
```
Comment (benisgold) [inline src/redux/gas.ts:369]: should be AvalanchePriceBumpFactor

File: src/utils/ethereumUtils.ts [hunk 4]
```diff
@@ -538,6 +529,8 @@ const getBasicSwapGasLimit = (chainId: number) => {
       return ethUnits.basic_swap_bsc;
     case getChainIdFromNetwork(Network.optimism):
       return ethUnits.basic_swap_optimism;
+    // check this out l8r
+    // case getChainIdFromNetwork(Network.avalanche):
```
Comment (benisgold) [inline src/utils/ethereumUtils.ts:533]: is this missing

File: src/model/remoteConfig.ts [hunk 5]
```diff
@@ -62,6 +66,7 @@ interface RainbowConfig extends Record<string, string | boolean | number> {
   op_chains_tx_enabled: boolean;
   mainnet_tx_enabled: boolean;
   goerli_tx_enabled: boolean;
+  avax_tx_enabled: boolean;
```
Comment (walmat) [inline src/model/remoteConfig.ts:69]: smol nit but i feel like this should be `avalanche_mainnet_rpc`
```suggestion
  avalanche_tx_enabled: boolean;
```

File: src/model/remoteConfig.ts [hunk 6]
```diff
@@ -52,6 +55,7 @@ interface RainbowConfig extends Record<string, string | boolean | number> {
   op_chains_enabled: boolean;
   mainnet_enabled: boolean;
   goerli_enabled: boolean;
+  avax_enabled: boolean;
```
Comment (walmat) [inline src/model/remoteConfig.ts:58]: smol nit but feel like this should be `avalanche_enabled`
```suggestion
  avalanche_enabled: boolean;
```

File: src/networks/avalanche.ts [hunk 7]
```diff
@@ -0,0 +1,68 @@
...
+import { getRemoteConfig } from '@/model/remoteConfig';
+
+export const getAvalancheNetworkObject = (): NetworkProperties => {
+  const { avax_enabled, avax_tx_enabled } = getRemoteConfig();
+  return {
+    // wagmi chain data
+    ...avalanche,
+
+    // network related data
+    enabled: avax_enabled || __DEV__,
```
Comment (walmat) [inline src/networks/avalanche.ts:16]: if you decide to switch `avax` -> `avalanche` it'll need to be updated here

File: src/networks/avalanche.ts [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+
+    // features
+    features: {
+      txHistory: true,
+      flashbots: false,
+      walletconnect: true,
+      swaps: true,
+      nfts: true,
+      pools: false,
+      txs: avax_tx_enabled || __DEV__,
```
Comment (walmat) [inline src/networks/avalanche.ts:41]: if you decide to switch `avax` -> `avalanche` it'll need to be updated here

File: src/redux/gas.ts (hunk 3 above)
Comment (walmat) [reply src/redux/gas.ts:369]: +1 keeping the naming consistent

**Review by walmat — COMMENTED** [review]
Same questions as ben. Also would like to keep naming consistent. Stick with avalanche unless referring to the symbol (AVAX)

File: src/networks/avalanche.ts (hunk 7 above)
Comment (skylarbarrera) [reply src/networks/avalanche.ts:16]: can we also remove dev here? doesnt match the others

File: src/utils/ethereumUtils.ts (hunk 4 above)
Comment (dereknelson) [reply src/utils/ethereumUtils.ts:533]: bx doesn't have it, removed

**Review by walmat — APPROVED** [review]

**Review by skylarbarrera — APPROVED** [review]
agree with smol nits but 🚀

**Review by ibrahimtaveras00 — APPROVED** [review]
Fixes look good.  Only remaining issue is APP-1220, but that seems to be out of scope for this ticket, so giving this a ✅

**Pull request merged**